src/charm.py

```diff
@@ -100,6 +100,8 @@ def __init__(self, *args):
             redis_relation={},
         )
         self._require_nginx_route()
+        self._gems_installed = False
```

Comment:
This variable is going to be set to `False` every time the charm class is initialised, i.e. on every independent hook invocation, so I think that means this will get run on every config-changed hook?

Reply:
I removed it and now the config-chaged uses a method to check if there is something on the /var/lib/gems to decide if will install them or not. I'm considering removing this check from the on_install event and assuming it will be installed/reinstalled. What do you think?